src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,51 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked")
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_CAPACITY = 10;
+    private final K[] keys = (K[]) new Object[ARRAY_CAPACITY];
+    private final V[] values = (V[]) new Object[ARRAY_CAPACITY];
+
     @Override
     public void put(K key, V value) {
+        int existingKeyIndex = findKey(key);
+        int size = size();
+        if (existingKeyIndex != -1) {
+            this.values[existingKeyIndex] = value;
+        } else if (size < ARRAY_CAPACITY) {
+            this.keys[size] = key;
+            this.values[size] = value;
+        } else {
+            throw new RuntimeException("Storage is full!");
+        }
```

Comment:
```suggestion
    if (size == ARRAY_CAPACITY) {
      throw new RuntimeException("Storage is full!");
    }
    int existingKeyIndex = findKey(key);
    if (existingKeyIndex != -1) {
      this.values[existingKeyIndex] = value;
      return;
    }
    this.keys[size] = key;
    this.values[size] = value;
    size++;
```

it's hart to read if-else constructions, better to avoid them where we can